src/main/java/org/ossgang/commons/observables/weak/WeakMethodReferenceObserver.java

```diff
@@ -48,7 +56,18 @@
  * subscription will be kept alive.
  */
 class WeakMethodReferenceObserver<C, T> implements Observer<T> {
-    private final WeakReference<C> holder;
+
+    private static final ScheduledExecutorService REFERENCE_CLEANUP_EXECUTOR = newSingleThreadScheduledExecutor(
+            daemonThreadFactoryWithPrefix("ossgang-weak-observer-cleanup"));
+    private static final Set<WeakCleaner> REFERENCE_CLEANERS = Collections.newSetFromMap(new ConcurrentHashMap<>());
+
+    static {
+        REFERENCE_CLEANUP_EXECUTOR.scheduleAtFixedRate(
+                () -> REFERENCE_CLEANERS.removeIf(WeakCleaner::attemptToClean),
+                1, 1, TimeUnit.SECONDS);
```

Comment:
We have to decide this rate. Could we make it parametrizable via sys properties? A sensible default could be like 30sec..